src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,150 @@
+package duke.util;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import duke.exception.DukeNoSuchTaskException;
+import duke.task.Task;
+
+/**
+ * TaskList class represents a list of tasks.
+ *
+ * @author Chng Zi Hao
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+    private Storage storage;
+
+    /**
+     * Constructor for TaskList.
+     *
+     * @param list List of task.
+     * @param storage Storage for Duke.
+     */
+    public TaskList(ArrayList<Task> list, Storage storage) {
+        this.list = list;
+        this.storage = storage;
+    }
+
+    /**
+     * Process user input and adds task that user has given and formats the message specific to the task.
+     *
+     * @param task The task user has specified
+     * @return Returns formatted string to be printed out to show task has been added or not.
+     */
+    public String addTask(Task task) {
+        list.add(task);
+        String totalTask = String.format("Now you have %d task(s) in the list.", list.size());
+        return String.format("Got it! I've added this task:\n  %s\n%s", list.get(list.size() - 1), totalTask);
+    }
+
+    /**
+     * Prints out the task list and shows all the task user has added in order.
+     */
+    public String printTaskList() {
+        if (list.size() == 0) {
+            return "List is empty. Add something to the list!\n";
+        }
+        String ls = "Here are the tasks in your list:\n";
```

Comment:
Might want to use a more descriptive name here? But I also get that it's only for a short section of the method.